packages/shared/hooks/useAsync.ts

```diff
@@ -107,6 +140,28 @@ export function useAsync<R, T extends Function>(cb?: AsyncCb<R, T>) {
   return [state, run, setAttempt] as const;
 }
 
+function useIsMounted() {
+  const isMounted = useRef(false);
+
+  useEffect(() => {
+    isMounted.current = true;
+
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+
+  return useCallback(() => isMounted.current, []);
+}
+
+// TODO: Rename the error.
```

Comment:
We may better rename this to CanceledError since it has nothing to do with Abort signal